src/sentry/sentry_metrics/querying/units.py

```diff
@@ -85,6 +90,36 @@ class UnitsSpec:
     units: Sequence[Unit]
 
 
+@dataclass(frozen=True)
+class UnitMetadata:
+    pass
+
+
+@dataclass(frozen=True)
+class WithNoUnit(UnitMetadata):
+    pass
+
+
+@dataclass(frozen=True)
+class WithFutureUnit(UnitMetadata):
+    pass
+
+
+@dataclass(frozen=True)
+class WithUnit(UnitMetadata):
+    unit_family: UnitFamily
+    reference_unit: str
+    unit: Unit
+    from_formula: bool = False
+
+    @property
+    def scaling_factor(self) -> int | float | None:
```

Comment:
The scaling factor should be returned only for non-formulas, since a formula has multiple scaling factors applied.